### DISCUSSION TITLE
Flexible Bar Duration? Allow a Bar to be Too Long or Too Short while Editing

### DISCUSSION THREAD

**Discussion by j-t-m-e** [helge17/tuxguitar#578] [General]
I haven't been using TuxGuitar for long, but I notice when composing, as soon as I enter a note it automatically fills the rest of the bar with rests. I see how this could be helpful, but this makes it difficult to edit the duration of notes after the fact because I find myself struggling with the strict limitation of not letting the bar be anything but exactly full.

Maybe I learned this workflow from years of using Guitar Pro, it's an option GP has always had that I missed right away.

Does this feature already exist in TuxGuitar? Can I disable strict bar duration somewhere?

If it doesn't exist, would it be reasonable to request this feature?

Here are some screenshots from Guitar Pro that hopefully help demonstrate what I've tried to describe:

Allow a bar to be too short or long during editing:
![gp duration1](https://github.com/user-attachments/assets/32c74330-7a6b-488e-bd38-36ed9bc85139)
![bar current duration too short](https://github.com/user-attachments/assets/9be2f23e-1a2e-491c-883b-14d21f5ced89)
![gp duration2](https://github.com/user-attachments/assets/69e74919-079f-45cb-9f9d-e0f1f52a96a7)
![bar current duration too long](https://github.com/user-attachments/assets/978409fb-453e-47a7-a3f4-2192083165ce)

A million thank yous from the bottom of my heart for open source software!

**Comment by guiv42** [comment]
Hi

> Does this feature already exist in TuxGuitar? Can I disable strict bar duration somewhere?

No, currently it's not possible.

> If it doesn't exist, would it be reasonable to request this feature?

Yes, such a feature request perfectly makes sense. I thought there was one already but I can't find it, so feel free to create a dedicated issue.
However, [measure manager](https://github.com/helge17/tuxguitar/blob/master/common/TuxGuitar-lib/src/org/herac/tuxguitar/song/managers/TGMeasureManager.java) is a part of code I haven't looked at deeply, and it's definitely not trivial. I believe that forcing all measures to be permanently valid greatly simplifies the job from software point of view (only the "nominal" mode needs to be managed). In other words, I have no precise idea of the feasibility of such an evolution.

**Reply by j-t-m-e** [reply]
Thank you so much for the reply! I don't know anything about programming but I had a hunch it wouldn't be trivial to implement, but wanted to bring it up to know for sure.

**Comment by guiv42** [comment]
> I thought there was one already but I can't find it

It's there: #226 (a discussion, not an issue)

**Comment by guiv42** [comment]
tracked by #579
